compiler/optimizations/gpuTransforms.cpp

```diff
@@ -1311,3 +1395,114 @@ void gpuTransforms() {
     }
   }
 }
+
+bool isLoopGpuBound(CForLoop* loop) {
+  return eligibleLoops.find(loop) != eligibleLoops.end();
+}
+
+// TODO: is LICM required for some loops to be eligible? thonk.
```

Comment:
As per my previous comment, it is imaginable that that may be the case. We wish that to help in distributed array support as well. 

Per distributed array access, we index into a per-locale table maintained by the runtime. That table is on the CPU memory. The value that we care in that table is the `.instance` of the array and shouldn't change. I think today that segfaults during kernel execution, ideally we should make distributed array accesses in kernels thwart GPU transformations. I am holding my breath that we'll fix the issue soon. But maybe not. Hoisting that is probably more difficult than relaxing LICM, and may require something more surgical.